File: client/lib/models/member_profile.dart
class MemberProfile {
  MemberProfile({
    required this.userId,
    required this.username,
    required this.displayName,
    required this.phoneNumber,
    required this.walletBalance,
  });

  final String userId;
  final String username;
  final String displayName;
  final String phoneNumber;
  final double walletBalance;

  factory MemberProfile.fromJson(Map<String, dynamic> json) {
    return MemberProfile(
      userId: json["userId"] as String,
      username: json["username"] as String,
      displayName: json["displayName"] as String,
      phoneNumber: json["phoneNumber"] as String,
      walletBalance: (json["walletBalance"] as num).toDouble(),
    );
  }
}

File: server/src/Lucky5.Domain/Entities/MemberProfile.cs
namespace Lucky5.Domain.Entities;

public sealed class MemberProfile
{
    public Guid UserId { get; init; }
    public string Username { get; init; } = string.Empty;
    public string DisplayName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public decimal WalletBalance { get; set; }
    public decimal SessionNetLoss { get; set; } // Tracks cumulative loss for pity timer (SuperBonus)
    public DateTime LastSeenUtc { get; set; } = DateTime.UtcNow;
}
